alpha/apps/kaltura/lib/myEntryUtils.class.php

```diff
@@ -895,7 +908,57 @@ public static function captureLocalThumb($entry, $capturedThumbPath, $calc_vid_s
 		myFileConverter::autoCaptureFrame($entry_data_path, $capturedThumbPath."temp_", $calc_vid_sec, -1, -1);
 		return true;
 	}
-	
+
+	public static function captureLocalThumbUsingPackager($entry, $capturedThumbPath, $calc_vid_sec, &$flavorAssetId)
+	{
+		$packagerCaptureUrl = kConf::get('packager_local_thumb_capture_url', 'local', null);
+		if (!$packagerCaptureUrl)
+			return false;
+
+		//look for the highest bitrate flavor tagged with thumbsource
+		$flavorAsset = assetPeer::retrieveHighestBitrateByEntryId($entry->getId(), flavorParams::TAG_THUMBSOURCE);
+
+		if(is_null($flavorAsset) || !self::isFlavorSupportedByPackager($flavorAsset))
+		{
+			// look for the highest bitrate flavor the packager can parse
+			$flavorAsset = assetPeer::retrieveHighestBitrateByEntryId($entry->getId(), flavorParams::TAG_MBR);
+			if (is_null($flavorAsset))
+			{
+				//retrieve original ready
+				$flavorAsset = assetPeer::retrieveOriginalReadyByEntryId($entry->getId());
+				if(is_null($flavorAsset) || !self::isFlavorSupportedByPackager($flavorAsset))
+					return false;
+			}
+		}
```

Comment:
Shoud be a private function that returns flavorAsset that is supported by packager